test/app.spec.js

```diff
@@ -0,0 +1,216 @@
+        .expect(400, done)
+    })
+  })
+
+  describe('POST /api/v1/athletes with last name is missing', () => {
+    it('should return a 400 error', done => {
+      request(app)
+        .post('/api/v1/athletes')
+        .send(Object.assign({}, athlete, { lastName: undefined }))
+        .set('Accept', 'application/json')
+        .expect(400, done)
+    })
+  })
+
+  describe('POST /api/v1/athletes with GHIN number too short', () => {
+    it('should return a 400 error', done => {
+      request(app)
+        .post('/api/v1/athletes')
+        .send(Object.assign({}, athlete, { ghinNumber: '123456' }))
+        .set('Accept', 'application/json')
+        .expect(400, done)
+    })
+  })
+
+  describe('GET /api/v1/athletes', () => {
+    it('should return a list of all athletes', done => {
+      request(app)
+        .get('/api/v1/athletes')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          expect(res.body).to.be.an('array')
+          done()
+        })
+    })
+  })
+
+  describe('GET /api/v1/athletes/:id', () => {
+    it('should return one athlete by id', done => {
+      request(app)
+        .get(`/api/v1/athletes/${id}`)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          expect(res.body.id).to.equal(id)
+          expect(res.body.lastName).to.equal('McTesterson')
+          done()
+        })
+    })
+  })
+
+  describe('GET /api/v1/athletes/:id with invalid id', () => {
+    it('should return a 404 error', done => {
+      request(app)
+        .get(`/api/v1/athletes/foo`)
+        .expect('Content-Type', /json/)
+        .expect(404, done)
+    })
+  })
+
+  describe('PUT /api/v1/athletes/:id', () => {
+    it('should update existing athlete by id', done => {
+      request(app)
+        .put(`/api/v1/athletes/${id}`)
+        .send(Object.assign({}, athlete, { id: id, firstName: 'Tester', createdAt: '1999-12-31' }))
+        .set('Accept', 'application/json')
+        .expect(204, done)
+    })
+  })
+
+  describe('GET /api/v1/athletes/:id', () => {
+    it('should return one updated athlete by id', done => {
+      request(app)
+        .get(`/api/v1/athletes/${id}`)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          expect(res.body.id).to.equal(id)
+          expect(res.body.firstName).to.equal('Tester')
+          done()
+        })
+    })
+  })
+
+  describe('PUT /api/v1/athletes/:id with invalid id', () => {
+    it('should return a 404 error', done => {
+      request(app)
+        .put(`/api/v1/athletes/foo`)
+        .expect('Content-Type', /json/)
+        .expect(404, done)
+    })
+  })
+
+  describe('PUT /api/v1/athletes/:id with last name missing', () => {
+    it('should return 400', done => {
+      request(app)
+        .put(`/api/v1/athletes/${id}`)
+        .send(Object.assign({}, athlete, { id: id, lastName: undefined }))
```

Comment:
You can use spread operator here:
```.send({ ...athlete, id, lastName: undefined })```

Reply:
Thanks for this tip. Much more concise. Will make these changes as well.